cmd/root.go

```diff
@@ -335,3 +350,51 @@ func initCertFlags() {
 		"Bearer token for authentication to the API server",
 	)
 }
+
+func initK8sFlagCompletion() {
+	_ = rootCmd.RegisterFlagCompletionFunc("context", k8sFlagCompletionFunc(func(cfg *api.Config) map[string]*api.Context {
+		return cfg.Contexts
+	}))
+
+	_ = rootCmd.RegisterFlagCompletionFunc("cluster", k8sFlagCompletionFunc(func(cfg *api.Config) map[string]*api.Cluster {
+		return cfg.Clusters
+	}))
+
+	_ = rootCmd.RegisterFlagCompletionFunc("user", k8sFlagCompletionFunc(func(cfg *api.Config) map[string]*api.AuthInfo {
+		return cfg.AuthInfos
+	}))
+
+	_ = rootCmd.RegisterFlagCompletionFunc("namespace", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		conn, err := client.InitConnection(client.NewConfig(k8sFlags))
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		nss, err := conn.ValidNamespaceNames()
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		return filterFlagCompletions(nss, toComplete)
+	})
+}
+
+func k8sFlagCompletionFunc[T any](picker func(cfg *api.Config) map[string]T) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		k8sCfg, err := client.NewConfig(k8sFlags).RawConfig()
```

Comment:
Should we pass in the k8s kubeconfig vs creating a new one for each instantiation?